pkg/helm/charts.go

```diff
@@ -95,12 +95,38 @@ func (c Charts) Vendor() error {
 	for _, r := range c.Manifest.Requires {
 		chartName := parseReqName(r.Chart)
 		chartPath := filepath.Join(dir, chartName)
-		if _, err := os.Stat(chartPath); err == nil {
-			log.Printf(" %s@%s exists", r.Chart, r.Version.String())
-			continue
+
+		_, err := os.Stat(chartPath)
+		if err == nil {
+			chartManifestPath := filepath.Join(chartPath, "Chart.yaml")
+			chartManifestBytes, err := ioutil.ReadFile(chartManifestPath)
+			if err != nil {
+				return fmt.Errorf("error reading chart manifest: %s", err)
+			}
+			var chartManifest map[string]interface{}
+			if err := yaml.Unmarshal(chartManifestBytes, &chartManifest); err != nil {
+				return fmt.Errorf("error unmarshalling chart manifest: %s", err)
+			}
+			currentChartVersion, ok := chartManifest["version"]
+			if !ok {
+				return fmt.Errorf("chart manifest %s does not contain a version field", chartManifestPath)
```

Comment:
Garbage in the charts folder would make us stop here, with on clear path to fix this.

Either we require manual user interaction (`rm -r` the folder, which we should suggest in that case), or we do it ourselves. 

wdyt? (also /cc @sdboyer, you probably have an opinion here :stuck_out_tongue_closed_eyes:)

Reply:
there are 3 error branches here that deal with garbage in the charts folder. On the one hand, we could consider the charts folder to be fully managed by tanka, and do what we want with it. On the other hand, it's always safer to not delete something when you don't know what it is 🙂 

I'm happy to defer to the maintainers on this.

Reply:
this error branch is now removed as a result of implementing the above request to create a struct for chartManifest, but I think this conversation still applies to the other branches